### PR TITLE
New monitoring metrics

### PR DESCRIPTION
**Pull request by sammsaski**
## Change description

> Adding total cpu metric capturing as opposed to per core, started recording system load average.

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

**Review by aywaldron — CHANGES_REQUESTED** [review]
Looks good Sam, just a couple small suggestions.

As for the failing builds, it looks like this is the redis key cleanup issue we've been tracking already. Try rerunning them to see if you can get them to pass. If not, we may have to experiment with adding a sleep somewhere in the tests to see if we can resolve this issue.

File: metrics/monitoring/metrics_monitor.py [hunk 1]
```diff
@@ -227,10 +248,34 @@ def cpu_metrics_update_cpu_stats(*keys):
...
+        # freq.current is calculated in MHz, so multiply by 10^6
+        #   so we can accurately label frequencies in dashboards
+        freq_current = freq.current * 1000000
         element.metrics_add(
-            cpu_freq_metrics_keys[str(i)], freq.current, pipeline=pipeline
+            cpu_freq_metrics_keys[str(i)], freq_current, pipeline=pipeline
         )
 
+    # Capture CPU frequencies wholistically
+    cpu_freq = psutil.cpu_freq().current * 1000000
```
Comment (ashule01) [inline metrics/monitoring/metrics_monitor.py:259]: NIT: similar comment to Anna's above about 10e6

File: metrics/monitoring/metrics_monitor.py [hunk 2]
```diff
@@ -227,10 +248,34 @@ def cpu_metrics_update_cpu_stats(*keys):
...
+    element.metrics_add(cpu_freq_metrics_keys[CPU_ALL], cpu_freq, pipeline=pipeline)
+
+    #
+    # CPU system metrics
+    #
+
+    # Returns system load averages over the last 1, 5, and 15 minutes as a tuple
+    #   We'll update using 1 minute average to get the highest level of accuracy
+    #   in the changes.
+    cpu_system_load_avg = [
```
[the remark this answers is not in the document]
Comment (sammsaski) [reply metrics/monitoring/metrics_monitor.py:269]: Yep, it's saving the average over the last minute upon calling. This isn't configurable when using `psutil` unfortunately. The title for the actual graph is marked with `(1 min)` to denote this, but it still needs to be updated in the performance stoplight section of the graph.

File: metrics/monitoring/metrics_monitor.py [hunk 3]
```diff
@@ -216,6 +234,9 @@ def cpu_metrics_update_cpu_stats(*keys):
     for i, pct in enumerate(cpu_percent):
         element.metrics_add(cpu_pct_metrics_keys[str(i)], pct, pipeline=pipeline)
 
+    cpu_percent = psutil.cpu_percent()
```
Comment (aywaldron) [inline metrics/monitoring/metrics_monitor.py:237]: Could you add comments clarifying this one is overall and the previous one with `percpu=True` is per core? documenting as much as possible will help out in the future. Same for the `cpu_freq` calls below

File: metrics/monitoring/metrics_monitor.py [hunk 4]
```diff
@@ -225,12 +249,38 @@ def cpu_metrics_update_cpu_stats(*keys):
...
+
+    #
+    # CPU system metrics
+    #
+
+    # Returns system load averages over the last 1, 5, and 15 minutes as a tuple
+    #   We'll update using 1 minute average to get the highest level of accuracy
+    #   in the changes.
+    cpu_system_load_avg = [
+        load_avg / psutil.cpu_count() * 100 for load_avg in psutil.getloadavg()
```
Comment (aywaldron) [inline metrics/monitoring/metrics_monitor.py:275]: Is the 100 here to make it a percentage? Would be helpful to document that in a comment as well.

**Review by dpipemazo — APPROVED** [review]
Thanks @sammsaski ! :tada:

**Pull request merged**